services/visual_recognition/v3.js

```diff
@@ -62,301 +37,329 @@ module.exports = function (RED) {
-        {
-          var err_desc = body.images[0].error.description;
-          var err_id = body.images[0].error.error_id;
+  function addTask (asyncTasks, msg, k, listParams, node) {
+    asyncTasks.push(function (callback) {
+      var buffer = msg.params[k];
+      temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
+        if (err) {
           node.status({fill:'red', shape:'ring', 
-                       text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_id']= err_id;
-          msg.result['error']= err_desc;
-          console.log('Error:', err_desc);
-          msg.payload='see msg.result.error';
-          node.send(msg); 
-        }
-        else {
-          msg.result = {};
-          msg.result['all'] = body;
-          msg.payload='see msg.result'; // to remove any Buffer that could remains
-          node.send(msg); 
-          node.status({});
+                       text:'unable to open image stream'});          
+          node.error('Node has been unable to open the image stream', msg);
+          return callback('open error on ' + k);
+        }  
+        stream_buffer(info.path, msg.params[k], function () {
+          listParams[k] = fs.createReadStream(info.path);
+          callback(null, k);
+        });
+      });
+    });
+  }
+
+  function prepareParamsCreateClassifier (params, node, msg, cb) {
+    var listParams = {}, asyncTasks = [] , k = null;
+    for (k in msg.params) {
+      if (k.indexOf('_examples') >= 0) {
+        addTask(asyncTasks, msg, k, listParams, node);
+      } else if (k === 'name') {
+        listParams[k] = msg.params[k];
+      }
+    } 
+    async.parallel(asyncTasks, function(error){
+      if (error) {
+        throw error;
+      }
+      for (var p in listParams) {
+        if (!p) {
+          params[p] = listParams[p];
         }
-      }; // actionComplete    
+      }
+      cb();
+    });
+  }
 
-      var actionComplete2 = function(err, body, other) {
+  function performDeleteAllClassifiers(params, node, msg) {
+    node.service.listClassifiers(params, function(err, body) {
+      node.status({});
+      if (err) {
+        node.status({fill:'red', shape:'ring', 
+          text:'Delete All : call to listClassifiers failed'});
+        node.error(err, msg);   
+      } else {
+        // Array to hold async tasks
+        var asyncTasks = [], nbTodelete = 0, nbdeleted = 0;
+        nbTodelete = body.classifiers.length;
+        body.classifiers.forEach(function (aClassifier) {
+          asyncTasks.push(function (cb) {
+            var parms = {};
+
+            parms.classifier_id = aClassifier.classifier_id;
+            node.service.deleteClassifier(parms, function(err) {
+              if (err) {
+                node.error(err, msg);
+                return cb('error');
+              }
+              nbdeleted++;
+              cb(null,parms.classifier_id);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'classifier_id' is not in camel case.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2737057989/issues/source?bid=3330694&fileBranchId=3352492#l241)